crates/kira/src/backend/cpal/desktop/stream_manager.rs

```diff
@@ -100,20 +87,9 @@ impl StreamManager {
 	}
 
 	/// Restarts the stream if the audio device gets disconnected.
-	fn check_stream(&mut self, stream_error_buffer: &mut Vec<StreamError>) {
-		if let State::Running {
-			stream_error_consumer,
-			..
-		} = &mut self.state
-		{
-			stream_error_buffer.append(
-				&mut stream_error_consumer
-					.queue
-					.lock()
-					.expect("Audio thread panicked while sending error"),
-			);
-
-			for error in stream_error_buffer.drain(..) {
+	fn check_stream(&mut self, stream_error_consumer: &mut Consumer<StreamError>) {
```

Comment:
Note that because we iterate over consumer now, I made it an argument instead so there will be no lifetime issues regarding method calls on self during error handling.